contracts/Aavegotchi/libraries/LibDelegatedWearables.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import {ItemRolesInfo, GotchiEquippedDepositsInfo, LibAppStorage, AppStorage} from "../libraries/LibAppStorage.sol";
+import {LibItems} from "../libraries/LibItems.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {LibERC1155} from "../../shared/libraries/LibERC1155.sol";
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+
+library LibDelegatedWearables {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    /**
+     * @notice Removes a delegated wearable from an Aavegotchi
+     * @dev Transfers the wearable from the Aavegotchi to the diamond and updates delegation info
+     * @param _slot The slot to remove the wearable from
+     * @param _gotchiId The id of the Aavegotchi
+     * @param _existingEquippedWearableId The id of the equipped wearable to be removed
+     */
+    function removeDelegatedWearableFromGotchi(uint256 _slot, uint256 _gotchiId, uint256 _existingEquippedWearableId) internal {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+
+        LibItems.removeFromParent(address(this), _gotchiId, _existingEquippedWearableId, 1);
```

Comment:
Why aren't we emitting the TransferSingle event here? @ernanirst

Reply:
Specifically, this event:

    IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(sender, address(this), sender, existingEquippedWearableId, 1);

Reply:
Is the answer no because the wearable is being unequipped from the Gotchi, but it's not being transferred from AavegotchiDiamond?